### PR TITLE
8285497: Add system property for Java SE specification maintenance version

### PR DESCRIPTION
**Pull request by justin-curtis-lu**
This PR adds the system property: _java.specification.maintenance.version_ with a value of 2. This change is done towards the Reference Implementation of the proposed Java SE 11 Maintenance Release 2 (https://mail.openjdk.org/pipermail/jdk8u-dev/2023-February/016710.html).

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8285497](https://bugs.openjdk.org/browse/JDK-8285497): Add system property for Java SE specification maintenance version

### Reviewers
 * [Lance Andersen](https://openjdk.org/census#lancea) (@LanceAndersen - **Reviewer**) ⚠️ Review applies to [b238cbb7](https://git.openjdk.org/jdk11u-ri/pull/5/files/b238cbb70b5df96fed98f52faeab9d0287c552a6)
 * [Joe Darcy](https://openjdk.org/census#darcy) (@jddarcy - **Reviewer**) ⚠️ Review applies to [b238cbb7](https://git.openjdk.org/jdk11u-ri/pull/5/files/b238cbb70b5df96fed98f52faeab9d0287c552a6)
 * [Naoto Sato](https://openjdk.org/census#naoto) (@naotoj - **Reviewer**) ⚠️ Review applies to [b238cbb7](https://git.openjdk.org/jdk11u-ri/pull/5/files/b238cbb70b5df96fed98f52faeab9d0287c552a6)
 * [Iris Clark](https://openjdk.org/census#iris) (@irisclark - **Reviewer**)
 * [Christoph Langer](https://openjdk.org/census#clanger) (@RealCLanger - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk11u-ri pull/5/head:pull/5` \
`$ git checkout pull/5`

Update a local copy of the PR: \
`$ git checkout pull/5` \
`$ git pull https://git.openjdk.org/jdk11u-ri pull/5/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 5`

View PR using the GUI difftool: \
`$ git pr show -t 5`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk11u-ri/pull/5.diff">https://git.openjdk.org/jdk11u-ri/pull/5.diff</a>

</details>

### CODE REVIEW

**Review by LanceAndersen — APPROVED** [review]
Looks good and in line with the previous PR in this area

**Review by jddarcy — APPROVED** [review]
If backported change is the same from a previous year and is made unchanged, the copyright year should match the year the original change was authored. Otherwise, looks fine.

**Review by naotoj — APPROVED** [review]

**Review by irisclark — APPROVED** [review]

File: src/java.base/share/classes/sun/security/provider/PolicyFile.java [hunk 1]
```diff
@@ -608,6 +608,9 @@ public Void run() {
                 pe.add(new PropertyPermission
                                 ("java.specification.version",
                                     SecurityConstants.PROPERTY_READ_ACTION));
+                pe.add(new PropertyPermission
+                        ("java.specification.maintenance.version",
```
Comment (RealCLanger) [inline src/java.base/share/classes/sun/security/provider/PolicyFile.java]: the indentation is not in sync with the lines above and below, maybe you could fix this.

File: src/java.base/share/conf/security/java.policy [hunk 2]
```diff
@@ -30,6 +30,8 @@ grant {
     permission java.util.PropertyPermission "line.separator", "read";
     permission java.util.PropertyPermission
                    "java.specification.version", "read";
+    permission java.util.PropertyPermission
+                       "java.specification.maintenance.version", "read";
```
Comment (RealCLanger) [inline src/java.base/share/conf/security/java.policy]: Same here, indentation could be fixed so as to be in sync with line 32.

File: src/java.base/share/native/libjava/System.c [hunk 3]
```diff
@@ -206,6 +206,8 @@ Java_java_lang_System_initProperties(JNIEnv *env, jclass cla, jobject props)
 
     PUTPROP(props, "java.specification.version",
             VERSION_SPECIFICATION);
+    PUTPROP(props, "java.specification.maintenance.version",
+                    "2");
```
Comment (RealCLanger) [inline src/java.base/share/native/libjava/System.c]: Same here, please check rows above and below.

**Review by RealCLanger — APPROVED** [review]

File: src/java.base/share/native/libjava/System.c (hunk 3 above)
Comment (justin-curtis-lu) [reply src/java.base/share/native/libjava/System.c]: Thank you for catching that, fixed the indentation for this and your previous comments.

**Review by RealCLanger — APPROVED** [review]

File: src/java.base/share/native/libjava/System.c (hunk 3 above)
Comment (RealCLanger) [reply src/java.base/share/native/libjava/System.c]: Great, thanks for updating. Looks nice now. 😄

**Review by irisclark — APPROVED** [review]

**Pull request closed without merging**